src/test/java/com/google/spanner/liquibase/LiquibaseTests.java

```diff
@@ -381,263 +388,241 @@ void doRealSpannerLoadUpdateDataTest() throws Exception {
-        new ColDesc("ColDateTime", "TIMESTAMP"),
-        new ColDesc("ColTimestamp", "TIMESTAMP"),
-        new ColDesc("ColDate", "DATE"),
-        new ColDesc("ColDecimal", "NUMERIC"),
-        new ColDesc("ColDouble", "FLOAT64"),
-        new ColDesc("ColFloat", "FLOAT64"),
-        new ColDesc("ColInt", "INT64"),
-        new ColDesc("ColMediumInt", "INT64"),
-        new ColDesc("ColNumber", "NUMERIC"),
-        new ColDesc("ColSmallInt", "INT64"),
-        new ColDesc("ColTime", "TIMESTAMP"),
-        new ColDesc("ColTinyInt", "INT64"),
-        new ColDesc("ColUUID", "STRING(36)"),
-        new ColDesc("ColXml", "STRING(MAX)"),
-        new ColDesc("ColBoolArray", "ARRAY<BOOL>"),
-        new ColDesc("ColBytesArray", "ARRAY<BYTES(100)>"),
-        new ColDesc("ColBytesMaxArray", "ARRAY<BYTES(MAX)>"),
-        new ColDesc("ColDateArray", "ARRAY<DATE>"),
-        new ColDesc("ColFloat64Array", "ARRAY<FLOAT64>"),
-        new ColDesc("ColInt64Array", "ARRAY<INT64>"),
-        new ColDesc("ColNumericArray", "ARRAY<NUMERIC>"),
-        new ColDesc("ColStringArray", "ARRAY<STRING(100)>"),
-        new ColDesc("ColStringMaxArray", "ARRAY<STRING(MAX)>"),
-        new ColDesc("ColTimestampArray", "ARRAY<TIMESTAMP>")
-    };
-    testTableColumns(testHarness.getJDBCConnection(), "TableWithAllLiquibaseTypes", cols);
-
-    testTablePrimaryKeys(testHarness.getJDBCConnection(), "TableWithAllLiquibaseTypes",
-        new ColDesc("ColBigInt"));
-    
-    // Generate a snapshot of the database.
-    SnapshotGeneratorFactory factory = SnapshotGeneratorFactory.getInstance();
-    CatalogAndSchema schema = new CatalogAndSchema("", "");
-    SnapshotControl control = new SnapshotControl(liquibase.getDatabase());
-    DatabaseSnapshot snapshot = factory.createSnapshot(schema, liquibase.getDatabase(), control);
-
-    testSnapshotTableAndColumns(snapshot, "TableWithAllLiquibaseTypes", cols);
-    testSnapshotPrimaryKey(snapshot, "TableWithAllLiquibaseTypes",
-        new ColDesc("ColBigInt", "INT64", Boolean.FALSE));
-    
-    // Generate an initial changelog for the database.
-    File changeLogFile = File.createTempFile("test-changelog", ".xml");
-    changeLogFile.deleteOnExit();
-    Main.run(new String[] {
-        "--overwriteOutputFile=true",
-        String.format("--changeLogFile=%s", changeLogFile.getAbsolutePath()),
-        String.format("--url=%s", testHarness.getJDBCConnection().getMetaData().getURL()),
-        "generateChangeLog"});
-    DocumentBuilderFactory documentBuilderFactory =
-        DocumentBuilderFactory.newInstance();
-    DocumentBuilder builder = documentBuilderFactory.newDocumentBuilder();
-    Document document = builder.parse(changeLogFile);
-    testTableColumnsInXml(document, "TableWithAllLiquibaseTypes", cols);
-
-    // Do rollback
-    liquibase.rollback(1, null);
-
-    // Ensure nothing is there!
-    testTableColumns(testHarness.getJDBCConnection(), "TableWithAllLiquibaseTypes");
+    try (Liquibase liquibase =
```

Comment:
`Liquibase` implements `AutoCloseable` and should therefore normally be wrapped in a [try-with-resources](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) block (or otherwise closed when the application is done with it) to let it clean up any resources it is using.